target/linux/bcm63xx/dts/bcm6328-adb-pdg-a4101n-a-000-1a1-ae.dts

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "bcm6328.dtsi"
+
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "ADB P.DG A4101N A-000-1A1-AE";
+	compatible = "adb,pdg-a4101n-a-000-1a1-ae", "brcm,bcm6328";
+
+	aliases {
+		led-boot = &led_dsl_green;
+		led-failsafe = &led_dsl_green;
+		led-running = &led_dsl_green;
+		led-upgrade = &led_dsl_green;
+	};
+
+	chosen {
+		bootargs = "rootfstype=squashfs,jffs2 noinitrd console=ttyS0,115200";
+		stdout-path = "serial0:115200n8";
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+
+		unnamed {
+			label = "unnamed_button";
+			gpios = <&pinctrl 12 1>;
+			linux,code = <KEY_WLAN>;
+			debounce-interval = <60>;
+		};
+
+		reset {
+			label = "reset";
+			gpios = <&pinctrl 23 1>;
+			linux,code = <KEY_RESTART>;
+			debounce-interval = <60>;
+		};
+
+		wps {
+			label = "wifi-led";
+			gpios = <&pinctrl 24 1>;
+			linux,code = <KEY_WPS_BUTTON>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&leds {
+	status = "okay";
+
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_leds>;
+
+	led@2 {
+		reg = <2>;
+		active-low;
+		label = "internet_led_enable";
```

Comment:
If this enables and disables the internet LED, maybe we should add it as a GPIO enabled by default, not as a LED.
You should also add a gpio-leds group with value 31 for the TEL LED since GPIO31 can't be controlled with the LED driver. See https://github.com/openwrt/openwrt/blob/master/target/linux/bcm63xx/dts/bcm6362-netgear-dgnd3700-v2.dts#L52